### PR TITLE
feat(cli): Print 'help msg' when the file or workflow parameters are …

### PR DESCRIPTION
**Pull request by gosoon**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [ ] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not.

### CODE REVIEW

File: cmd/argo/commands/lint.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ func NewLintCommand() *cobra.Command {
 		Use:   "lint FILE...",
 		Short: "validate files or directories of workflow manifests",
 		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
```
Comment (alexec) [inline cmd/argo/commands/lint.go:25]: this actually intentional - works with xargs

Comment (gosoon) [reply cmd/argo/commands/lint.go:25]: ok

Comment (gosoon) [reply cmd/argo/commands/lint.go:25]: It feels a little unreasonable to just output "Workflow manifests validated" without adding any files.
```
$ argo lint
Workflow manifests validated
```

Comment (gosoon) [reply cmd/argo/commands/lint.go:25]: kubectl use the `-f` parameter, If no file is specified, `pflag` will throw an error, and the output will print "flag needs an argument: 'f' in -f" and help information, exit code is 1.

`argo lint` director add file name, I think it is ok to directly output help information without specifying file, exit code is also 1. If need to be consistent with kubectl, we also need to use the `-f` parameter, which is a big change.

```
$ kubectl create -f
Error: flag needs an argument: 'f' in -f


Examples:
  # Create a pod using the data in pod.json.
  kubectl create -f ./pod.json
......



$ echo $?
1
```

Comment (alexec) [reply cmd/argo/commands/lint.go:25]: I'm not sure this is apples-and-apples. I guess the nearest would be `kubectl create --dry-run -f FILE`?

Either way, can you come up with a solution that works with `xargs`? We can't break existing users like this.

Comment (gosoon) [reply cmd/argo/commands/lint.go:25]: ok

**Pull request closed without merging**